crates/store/re_datafusion/src/grpc_streaming_provider.rs

```diff
@@ -0,0 +1,166 @@
+use std::{any::Any, pin::Pin, sync::Arc};
+
+use async_trait::async_trait;
+
+use arrow::{array::RecordBatch, datatypes::SchemaRef};
+use datafusion::{
+    catalog::{Session, TableProvider},
+    error::{DataFusionError, Result as DataFusionResult},
+    execution::{RecordBatchStream, SendableRecordBatchStream, TaskContext},
+    physical_plan::{
+        streaming::{PartitionStream, StreamingTableExec},
+        ExecutionPlan,
+    },
+    prelude::Expr,
+};
+use futures_util::StreamExt as _;
+use tokio_stream::Stream;
+
+#[async_trait]
+pub trait GrpcStreamToTable:
+    std::fmt::Debug + 'static + Send + Sync + Clone + std::marker::Unpin
+{
+    type GrpcStreamData;
+
+    async fn create_schema(&mut self) -> Result<SchemaRef, DataFusionError>;
```

Comment:
I'm also tripped up by this name. The schema is not created, since it exists _somewhere_ to begin with, right?. So more like `get_schema` or `fetch_schema` (I prefer the latter, indicates that the schema is to be retrieved by the grpc stream originating service).